she/evolve.py

```
@@ -0,0 +1,328 @@
+class Hypothesis:
+    """One competing repair hypothesis for a novel failure."""
+
+    kind: str
+    rationale: str
+    required_gates: tuple[str, ...]
+    isolated: bool = True
+    selected: bool = False
+
+    def __post_init__(self) -> None:
+        if self.kind not in HYPOTHESIS_KINDS:
+            raise EvolveError(f"unknown hypothesis kind: {self.kind!r}")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)